### ISSUE TITLE
Unable to remove site page URL

### ISSUE THREAD

**Issue by thabaum** [dnnsoftware/Dnn.Platform#3097]
## Description of bug
Page Settings > Advanced > SEO
Unable to edit or remove URLs for pages

## Screenshots
![site-alias-unable-to-remove](https://user-images.githubusercontent.com/13577556/44485055-4a370580-a604-11e8-9cc4-87992032d4ec.png)

## Affected version

* [x] 9.2.2
* [x] 9.2.1
* [x] 9.2
* [x] 9.1.1
* [ ] 9.1
* [ ] 9.0

## Affected browser

* [ ] Chrome
* [ ] Firefox
* [ ] Safari
* [ ] Internet Explorer
* [ ] Edge

**Comment by valadas** [comment]
Would anyone object to simply at least allow deleting the system generated ones and just check to ensure the user cannot delete the last Active (200) one. Does any one see a good reason to not allow modifying/deleting system-generated ones ?

**Comment by sleupold** [comment]
there should be for sure no duplicate entries, DNN should prevent adding it.

**Comment by jahalliv** [comment]
Has there been any work on this, very frustrating.

**Comment by valadas** [comment]
@jahalliv please consider this is a free open source product, no reason to be frustrated here. Developers work on what they decide to work on and there has been a lot of recent major activity. Also for issues like this that are not obvious to make happen. Writing very very clear steps to reproduce from scratch may help for someone to pick it up.

**Comment by mitchelsellers** [comment]
As I look at this, unless someone else has different recreation steps, I'm not 100% sure this is actually an issue.

The only way that I have been able to get multiple URL's with the same path, and not able to edit is with a site with individual languages.  This is expected, those are the default routes for those languages.  Maybe the language should be shown here, that could help.

Custom added ones can be removed. 

Renaming a page, results in a redirect from the old, but it can be removed if needed.

Does someone have solid recreation steps for this issue that can cause this issue without it being a multi language site?

**Comment by valadas** [comment]
I did happen to me but I do not have solid recreation steps. Time providing I would like to allow them to be deleted no matter what (as long as it is not the last active one). Like I said earlier, just did not have the time to implement that.

**Comment by jahalliv** [comment]
The issue description is just part of it In older versions of DNN you could add a page with a space in the name and then override the URL with a dash removing it. Can't do that anymore. When you try to add it you can't, DNN tells you its "unavailable". Turns out it was silently added as a damn automatic URL that can't be deleted.

**Comment by mitchelsellers** [comment]
I don't believe that removing the automatic behavior of dashes/not is something that we should be doing in the provider.  That would have huge implications for site behaviors.

Ideally, a concise recreation process with notes of what is expected is going to make this actionable.

**Comment by jahalliv** [comment]
Then at least give the site administrator the option to delete an automatically created redirect OR the option to change it to "active". The way it works now discourages the good practice of eliminating dashes from URLs.

**Comment by mitchelsellers** [comment]
@jahalliv I'm curious, why do you state that dashes should be removed?  The current consensus still supports the usage of dashes for SEO as it is treated as a word-break from a content perspective?

**Comment by mitchelsellers** [comment]
Straight from Google - https://developers.google.com/search/docs/crawling-indexing/url-structure

**Comment by jahalliv** [comment]
Mitch, Lets stay on topic and avoid the strawman. No one wants their platform dictating their design practice to justify NOT fixing the problem.

**Comment by valadas** [comment]
#5401 might be the way to make this problem happen

**Comment by mwigley-trilogy** [comment]
Yep, as you've noticed dashed and non-dashed URL's are added automatically for pages.

My fix for #5401 should fix all weird UI bugs that you've seen with lots of extra rows showing up.

In terms of being able to remove the 'automatic' URL's, that seems like a reasonable feature I guess I'd suggest a starting point of looking at
\Dnn.Platform\Dnn.AdminExperience\Dnn.PersonaBar.Extensions\Components\Pages\PageUrlsController.cs
GetPageUrls
and how it generates the 'automatic' ('system') Urls in this.GetSortedUrls.  The logic appears to get a wee bit complex though, and personally I'd be scared to mess with it too much.

**Comment by mitchelsellers** [comment]
Yes, all of these settings still work

The history on this is that when DNN Software Acquired UrlMaster years ago, the functionality was left, but the UI was removed from the OpenSource bits.
